src/Pate/Verification/StrongestPosts.hs

```diff
@@ -0,0 +1,917 @@
+  AbstractDomain sym arch {- ^ current abstract domain -} ->
+  PairGraph sym arch ->
+  PPa.PatchPair (PB.BlockTarget arch) {- ^ next entry point -} ->
+  EquivM sym arch (PairGraph sym arch)
+triageBlockTarget asm bundle currBlock d gr (PPa.PatchPair blktO blktP) =
+  withEmptyAssumptionFrame $
+  do let
+        blkO = PB.targetCall blktO
+        blkP = PB.targetCall blktP
+        pPair = PPa.PatchPair blkO blkP
+
+     traceBundle bundle ("  targetCall: " ++ show blkO)
+     withAssumption_ (return asm) $
+       -- TODO? use withSatAssumption here, or something similar using an online solver?
+       withAssumption_ (PD.matchesBlockTarget bundle blktO blktP) $
+
+       case (PB.targetReturn blktO, PB.targetReturn blktP) of
+         (Just blkRetO, Just blkRetP) ->
+           do traceBundle bundle ("  Return target " ++ show blkRetO ++ ", " ++ show blkRetP)
+
+              -- TODO, this isn't correct.  Syscalls don't correspond to
+              -- "ArchTermStmt" in any meaningful way, so this is all a misnomer.
+              isSyscall <- case (PB.concreteBlockEntry blkO, PB.concreteBlockEntry blkP) of
+                 (PB.BlockEntryPreArch, PB.BlockEntryPreArch) -> return True
+                 (entryO, entryP) | entryO == entryP -> return False
+                 _ -> throwHere $ PEE.BlockExitMismatch
+              traceBundle bundle ("  Is Syscall? " ++ show isSyscall)
+
+              ctx <- view PME.envCtxL
+              let isEquatedCallSite = any (PPa.matchEquatedAddress pPair) (PMC.equatedFunctions ctx)
+
+              isPLT <- findPLTSymbol blkO blkP
+
+              if | isSyscall -> handleSyscall bundle currBlock d gr pPair (PPa.PatchPair blkRetO blkRetP)
+                 | isEquatedCallSite -> handleInlineCallee bundle currBlock d gr pPair (PPa.PatchPair blkRetO blkRetP)
+                 | Just pltSymbol <- isPLT -> handlePLTStub bundle currBlock d gr pPair (PPa.PatchPair blkRetO blkRetP) pltSymbol
+                 | otherwise -> handleOrdinaryFunCall bundle currBlock d gr pPair (PPa.PatchPair blkRetO blkRetP)
+
+         (Nothing, Nothing) ->
+           do traceBundle bundle "No return target identified"
+              handleJump bundle currBlock d gr pPair
+
+         _ -> do traceBundle bundle "BlockExitMismatch"
+                 throwHere $ PEE.BlockExitMismatch
+
+-- | See if the given jump targets correspond to a PLT stub for
+--   the same symbol, and return it if so.
+findPLTSymbol ::
+  PB.ConcreteBlock arch PBi.Original ->
+  PB.ConcreteBlock arch PBi.Patched ->
+  EquivM sym arch (Maybe BS.ByteString)
+findPLTSymbol blkO blkP =
+  do PA.SomeValidArch archData <- asks envValidArch
+     let oExtraMap = PA.validArchOrigExtraSymbols archData
+     let pExtraMap = PA.validArchPatchedExtraSymbols archData
+
+     let oAddr = PAd.addrToMemAddr (PB.concreteAddress blkO)
+     let pAddr = PAd.addrToMemAddr (PB.concreteAddress blkP)
+
+     case (MM.asAbsoluteAddr oAddr, MM.asAbsoluteAddr pAddr) of
```

Comment:
I believe this will always fail for position-independent executables. Depending on what exactly you need here, this might be better as `MM.segmentOffAddr`

Reply:
I admit I don't fully understand how the data in the "extra symbols" map corresponds to what I have available here; if there's a better way to do this, I'd love to know.